### PR TITLE
Change theme settings default values to improve merchant UX

### PR DESCRIPTION
**Pull request by melissaperreault**
**TLDR; Why are these changes introduced?**

Change theme settings default values to reduce the amount of steps merchants go through to discover what theme settings are doing in the editor preview.

**What approach did you take?**

Applied the safe smart defaults where needed to follow this rule:

- If `Border thickness` default = `0`
   - Make the `Border opacity` default = `10`
      - EXCEPTION, if `Button's Border thickness` is set to `0`, make it to `1` and have the opacity to `100`, this will ensure the secondary button style outline is visible.
- If `Shadow opacity` default = `0`
   - Make `Shadow vertical offset` default = `4`
   - Make `Shadow blur` default = `5`

**Other considerations**

**Testing steps/scenarios**
- [ ] Stress test defaults are well applied and don't conflict with other styles/settings.
- [ ] Areas to stress test if this system works
   - [ ] Increase the Shadow opacity to notice the fewer steps it takes to preview the style
      - [ ] Buttons
      - [ ] Variant pills
      - [ ] Inputs
      - [ ] Cards
      - [ ] Content containers
      - [ ] Media
      - [ ] Dropdown and pop-ups
      - [ ] Drawers
   - [ ] Increase the border thickness to notice the fewer steps in previewing the border to appear
      - [ ] Cards
      - [ ] Content containers

**Demo links**

- [Editor](https://os2-demo.myshopify.com/admin/themes/127629459478/editor)

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: config/settings_schema.json [hunk 1]
```diff
@@ -558,7 +558,7 @@
         "step": 5,
         "unit": "%",
         "label": "t:settings_schema.global.settings.opacity.label",
-        "default": 10
+        "default": 0
```
Comment (melissaperreault) [inline config/settings_schema.json:561]: I wonder why this was set at 10?
